.github/workflows/deploy-github-pages.yml

```
@@ -0,0 +1,55 @@
+name: Deploy GitHub Pages
+on:
+  workflow_call:
+    inputs:
+      version:
+        description: "Version to prepare documentation for. Unset version will update 'latest' version."
+        required: false
+        type: string
+        default: ''
+
+jobs:
+  deploy-gh-pages:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Checkout github pages branch
+        uses: actions/checkout@v4
+        with:
+          path: gh-pages
+          ref: gh-pages
+
+      # v37.1.0
+      - uses: cachix/install-nix-action@d1ca217b388ee87b2507a9a93bf01368bde7cec2
+        with:
+          install_url: https://releases.nixos.org/nix/nix-2.23.3/install
+          github_access_token: "${{ secrets.GITHUB_TOKEN }}"
+
+      - name: Set Git credentials
+        run: |
+          git config --global 'url.https://api@github.com/'.insteadOf 'https://github.com/'
+          git config --global 'url.https://ssh@github.com/'.insteadOf 'ssh://git@github.com/'
+          git config --global 'url.https://git@github.com/'.insteadOf 'git@github.com:'
+       # v16
+      - uses: cachix/cachix-action@0fc020193b5a1fa3ac4575aa3a7d3aa6a35435ad
+        with:
+          name: veridise-public
+          authToken: '${{ secrets.CACHIX_PUBLIC_TOKEN }}'
+
+      - name: Build documentation
+        run: nix --print-build-logs build '.#docs'
+
+      - name: Move built documentation to latest
+        if: ${{ inputs.version == '' }}
+        run: mv result/html/* gh-pages
+
+      - name: Move documentation to version folder
+        if: ${{ inputs.version != '' }}
+        run: |
+          mkdir gh-pages/${{ inputs.version }}
+          mv result/html/* gh-pages/${{ inputs.version }}
```

The substitutions need to be quoted. I suggest saving the version to an env var like this:
```suggestion
        env:
          VERSION: ${{ inputs.version }}
        run: |
          mkdir gh-pages/"$VERSION"
          mv result/html/* gh-pages/"$VERSION"
```